lib/SPIRVProducerPass.cpp

```diff
@@ -377,6 +377,13 @@ struct SPIRVProducerPassImpl {
       HasVariablePointers = true;
     }
   }
+  bool hasNonUniformPointers() { return HasNonUniformPointers; }
+  void setNonUniformPointers() {
+    if (HasNonUniformPointers) {
+      addCapability(spv::CapabilityShaderNonUniform);
+      HasNonUniformPointers = true;
+    }
+  }
```

Comment:
This looks off. The control is set to true only if it is already true. Also `HasNonUniformPointers` doesn't seem to get initialized by the constructors.

Reply:
Fixed